### ISSUE TITLE
view_schema_test failed for memory leaks

### ISSUE THREAD

**Issue by amoskong** [scylladb/scylla#2939]
*Installation details*
Scylla version (or git commit hash): aeea83172
Cluster size: 1
OS (RHEL/CentOS/Ubuntu/AWS AMI): CentOS

The problem was introduced by http://jenkins.cloudius-systems.com:8080/job/scylla-master-build/compiler=g++,label=master/579/changes
```
eea83172 tests/cql_query_test: Add test case for secondary index queries
9048f741a cql3: Secondary-index backed select statements
3150962cb index: Fix index view schema when primary key component is indexed
3c9060798 tests/cql_query_test: Fix view creation in test_duration_restrictions()
c243a0c8f cql3/restrictions: Add statement_restrictions::index_restrictions() helper
678a6f6e2 index: Implement index::supports_expression() for EQ operator
04b482146 cql3: Make operator_type class non-copyable
1ae9343f6 index: Fix index::supports_expression() operator parameter type
3e3c580f7 cql3: Implement statement_restriction index validation
```
Backtrace:
  0x00000000006eda6a
  0x0000000000420ac3
  0x000000000042757d
  /lib64/libpthread.so.0+0x00000000000115cf
  0x00000000008d841d
  0x00000000008cde8f
  0x00000000008c139e
  0x00000000008b3826
  0x00000000008a7f8e
  0x000000000088f86c
  0x000000000087a21b
  0x00000000008b3671
  0x00000000008a7f8e
  0x000000000088f86c
  0x000000000088f8eb
  0x000000000087a21b
  0x00000000008592af
  0x0000000000823e25
  0x00000000007d713e
  0x000000000077daac
  0x00000000007270c2
  0x0000000004013b52
  0x00000000040c5d72
  0x00000000040aee82
  0x000000000487e610
  0x000000000487f167
  0x000000000378a547
  0x0000000003a54e59
  0x000000000399edde
  0x000000000399e271
  0x000000000488a601
  0x0000000004886ca4
  0x0000000008ed5a74
  0x000000000135a251
  0x00000000016b85a4
  0x0000000008f27210
  0x0000000008ebb226
  0x0000000008ecc010
  0x0000000008ecc096
  0x0000000008ecc125
  0x0000000008ec5518
  0x0000000008ec0b64
  0x0000000008ecf965
  0x0000000000719147
  0x0000000000aed650

=================================================================
==3978==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 160 byte(s) in 5 object(s) allocated from:
    #0 0x7fd91c4b2b30 in realloc (/lib64/libasan.so.3+0xdcb30)
    #1 0x7fd91ab38003  (/lib64/libstdc++.so.6+0x92003)

SUMMARY: AddressSanitizer: 160 byte(s) leaked in 5 allocation(s).

=== stdout END ===
```
@penberg

**Comment by amoskong** [comment]
Subtest log of view_schema_test:  [view_schema_test_debug.log.txt](https://github.com/scylladb/scylla/files/1448745/view_schema_test_debug.log.txt)
[Full console log of jenkins job 579](http://jenkins.cloudius-systems.com:8080/job/scylla-master-build/compiler=g++,label=master/579/consoleFull)

**Comment by duarten** [comment]
Fixed by 328b908574f86fd0dfafdb66da2947811b5a7d95 and b515cca5e2f3f1d3ccff563fc6b72cb9a6314687.
